odoo/modules/loading.py

```diff
@@ -171,6 +194,11 @@ def load_test(idref, mode):
             or hasattr(package, "update")
             or package.state in ("to install", "to upgrade")
         )
+
+        mode = 'update'
+        if hasattr(package, 'init') or package.state == 'to install':
+            mode = 'init'
+
```

Comment:
Might conflict with f61262eb08134ba5bdf9099445fe821769842bb3 where you only moved this below the `registry.load` call.